pkg/commands/copy.go

```diff
@@ -134,3 +135,46 @@ func (c *CopyCommand) FilesUsedFromContext(config *v1.Config, buildArgs *dockerf
 func (c *CopyCommand) MetadataOnly() bool {
 	return false
 }
+
+func (c *CopyCommand) RequiresUnpackedFS() bool {
+	return true
```

Comment:
This if for unpacking the base image, right? If so, I would guess we don't need it for the copy command (unless I'm completely forgetting something which is totally possible)

Reply:
It turns you absolutely need it if you are copying something into a directory that is expected to exist in the base image.

So, here's the logic for COPY:
- COPY fromsomething tosomething
   + if tosomething is a directory, or it ends with '/' then copy fromsomething INTO tosomething
   + if tosomething does not exist or it is a file, then copy fromsomething OVER tosomething

So, in the test case I added, the dockerfile says:
FROM blahblah

COPY context/foo /usr/bin

If the base image does not exist, then we copy the contents of context/foo into a file called /usr/bin, which is not intended, and not how 'docker build' behaves.

Reply:
ah makes sense, thanks!